### PR TITLE
Storybook v4

### PR DESCRIPTION
**Pull request by brandonellis**
This PR updates all the dependencies to work with Storybook v4.  Older version of this repo required components from the `@storybook/addon-info` which have changed quite a bit and some are no longer exported. Therefore I have extracted those components and added them directly to this repo.  There is one area of markdown info which still has a dependency but this branch is already getting too large. Tests pass.

### CODE REVIEW

**Comment by tweeres04** [comment]
Extracting the components that other libraries have deprecated is quite a smelly code smell. Is there no way we can use the APIs they've moved to?

**Comment by brandonellis** [comment]
@tweeres04 Yep. I don't disagree.  There are a few options that ran through my mind here of what we could do. 
1.  We separate ourselves from exporting components from another dependency which no longer allows exporting of said components.

File: .eslintrc [hunk 1]
```diff
@@ -7,12 +7,12 @@
...
+    "import/no-named-as-default": 0,
+    "import/no-named-as-default-member": 0,
+    "import/named": 0
   },
   "parserOptions": {
-    "ecmaVersion": 6,
-    "ecmaFeatures": {
-      "experimentalObjectRestSpread": true
-    }
+    "ecmaVersion": 6
```
Comment (tweeres04) [inline .eslintrc]: You should be good to kick this all the way up to ecmaVersion 8 or whatever the latest is

File: .scripts/mocha_runner.js [hunk 2]
```diff
@@ -4,15 +4,18 @@
...
 // Use `.scripts/user/pretest.js instead`.
 
-require('babel-core/register');
-require('babel-polyfill');
+require('@babel/register');
+require('@babel/polyfill');
 
 // Add jsdom support, which is required for enzyme.
-var jsdom = require('jsdom').jsdom;
+var jsdom = require('jsdom');
```
Comment (tweeres04) [inline .scripts/mocha_runner.js]: Could just do `const { JSDOM } = require('jsdom');`

But you should kill the `var` regardless

File: .storybook/webpack.config.js [hunk 3]
```diff
@@ -9,17 +9,12 @@ const updateConfig = require('./user/modify_webpack_config');
 
 const config = {
   module: {
-    loaders: [
+    rules: [
```
Comment (tweeres04) [inline .storybook/webpack.config.js]: Right, that's what this was

File: src/components/Props.jsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
```
Comment (tweeres04) [inline src/components/Props.jsx]: Do we need this file still?

**Comment by brandonellis** [comment]
Ok, this bad boy now uses an import statement (from the `@storybook/addon-info`) dependency for the components which were not working.  Used the direct path, which results in the correct functioning and no need to duplicate code or create potential code smells. 

It also fixes, 1 or 2 open issues listed by the community - which includes the console warnings around glamorous issues. 

[https://github.com/Checkfront/react-storybook-addon-chapters/issues/34](url)

**Pull request merged**